bridges/snowbridge/pallets/ethereum-client/src/tests_electra.rs

```diff
@@ -0,0 +1,926 @@
+				&finalized_update.clone(),
+				finalized_update.finalized_header.slot + free_headers_interval
+			),
+			Pays::Yes
+		);
+		// Is free, larger than the minimum interval
+		assert_eq!(
+			EthereumBeaconClient::check_refundable(
+				&finalized_update,
+				finalized_update.finalized_header.slot - (free_headers_interval + 2)
+			),
+			Pays::No
+		);
+		// Is free, valid sync committee update
+		assert_eq!(
+			EthereumBeaconClient::check_refundable(
+				&sync_committee_update,
+				finalized_update.finalized_header.slot
+			),
+			Pays::No
+		);
+	});
+}
+
+#[test]
+pub fn verify_merkle_branch_for_finalized_root() {
+	new_tester().execute_with(|| {
+		assert!(verify_merkle_branch(
+			hex!("0000000000000000000000000000000000000000000000000000000000000000").into(),
+			&[
+				hex!("0000000000000000000000000000000000000000000000000000000000000000").into(),
+				hex!("5f6f02af29218292d21a69b64a794a7c0873b3e0f54611972863706e8cbdf371").into(),
+				hex!("e7125ff9ab5a840c44bedb4731f440a405b44e15f2d1a89e27341b432fabe13d").into(),
+				hex!("002c1fe5bc0bd62db6f299a582f2a80a6d5748ccc82e7ed843eaf0ae0739f74a").into(),
+				hex!("d2dc4ba9fd4edff6716984136831e70a6b2e74fca27b8097a820cbbaa5a6e3c3").into(),
+				hex!("91f77a19d8afa4a08e81164bb2e570ecd10477b3b65c305566a6d2be88510584").into(),
+			],
+			subtree_index(crate::config::altair::FINALIZED_ROOT_INDEX),
+			generalized_index_length(crate::config::altair::FINALIZED_ROOT_INDEX),
+			hex!("e46559327592741956f6beaa0f52e49625eb85dce037a0bd2eff333c743b287f").into()
+		));
+	});
+}
+
+#[test]
+pub fn verify_merkle_branch_fails_if_depth_and_branch_dont_match() {
+	new_tester().execute_with(|| {
+		assert!(!verify_merkle_branch(
+			hex!("0000000000000000000000000000000000000000000000000000000000000000").into(),
+			&[
+				hex!("0000000000000000000000000000000000000000000000000000000000000000").into(),
+				hex!("5f6f02af29218292d21a69b64a794a7c0873b3e0f54611972863706e8cbdf371").into(),
+				hex!("e7125ff9ab5a840c44bedb4731f440a405b44e15f2d1a89e27341b432fabe13d").into(),
+			],
+			subtree_index(crate::config::altair::FINALIZED_ROOT_INDEX),
+			generalized_index_length(crate::config::altair::FINALIZED_ROOT_INDEX),
+			hex!("e46559327592741956f6beaa0f52e49625eb85dce037a0bd2eff333c743b287f").into()
+		));
+	});
+}
+
+#[test]
+pub fn sync_committee_participation_is_supermajority() {
+	let bits =
+		hex!("bffffffff7f1ffdfcfeffeffbfdffffbfffffdffffefefffdffff7f7ffff77fffdf7bff77ffdf7fffafffffff77fefffeff7effffffff5f7fedfffdfb6ddff7b"
+	);
+	let participation =
+		snowbridge_beacon_primitives::decompress_sync_committee_bits::<512, 64>(bits);
+	assert_ok!(EthereumBeaconClient::sync_committee_participation_is_supermajority(&participation));
+}
+
+#[test]
+pub fn sync_committee_participation_is_supermajority_errors_when_not_supermajority() {
+	new_tester().execute_with(|| {
+		let participation: [u8; 512] = [
+			0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 1, 1, 0, 0, 0, 0,
```

Comment:
not important, but would prefer this array to hex-encoded, as its more compact.

Reply:
Good point, addressed in https://github.com/paritytech/polkadot-sdk/pull/7075/commits/a25a8b834054b8770f58c1037490c81b94770a9b.